package.json

```diff
@@ -129,6 +130,7 @@
     "vips"
   ],
   "dependencies": {
+    "@emnapi/runtime": "^0.31.0",
```

Comment:
This will cause every `npm install sharp` to also install `@emnapi/runtime`, regardless of the user's intent to run wasm or native. Is that intended?

Reply:
Not necessarily, but, unfortunately, there is no good way to split dependencies by target (e.g. in the opposite direction we also don't need node-addon-api or detect-libc for the Wasm package, but can't exclude it).

The only way I can think of is installing those deps by hand as part of a postinstall script or something, but that would get very messy.

Reply:
typically, `peerDependencies` are used in such cases

Reply:
Another option is to publish as a separate package `sharp-wasm`. 

This is how esbuild does it https://www.npmjs.com/package/esbuild-wasm

Reply:
imho, what `styfle` proposes is the best way longterm, but it'd require having 3 packages to address what `RReverser` mentioned about useless `dependencies` for the `wasm` target.
- `sharp-core` to define common logic (and dependencies) between wasm/native
- `sharp-node` or `sharp` (the name might be affected by plans on adopting `bun` #3511 and vision on whether there'd be a difference between `bun` & `node` targets)
- `sharp-wasm`

Reply:
There's been some recent movement around (the previously-stalled) npm package `distributions`, which should help solve this problem - see https://github.com/npm/statusboard/issues/642

Reply:
I think `distributions` would be nice but the RFC isn't even approved yet so its still a long way off from being available in all major package managers (npm, yarn, pnpm, etc). Furthermore, it still works by installing separate packages so there is no harm in starting with a package exclusively for wasm now.

Reply:
Tbh the `@emnapi/runtime` is relatively small in comparison to everything else, especially the native module itself, so personally I'd be leaning towards just leaving it as-is for now. Yes, it will download a small unnecessary package, but the complexity of splitting different variants up seems a bit excessive.

Reply:
Totally agree with `RReverser`.
Separation into multiple packages, if there'd be users who depend on the installation size, should be done in a separate PR.